src/MainWindow.vala

```diff
@@ -95,6 +90,22 @@ public class AppCenter.MainWindow : Gtk.ApplicationWindow {
         homepage.subview_entered.connect (view_opened);
         installed_view.subview_entered.connect (view_opened);
         search_view.subview_entered.connect (view_opened);
+
+        NetworkMonitor.get_default ().network_changed.connect (update_view);
+
+        network_alert_view.action_activated.connect (() => {
+            var list = new List<string> ();
```

Comment:
This should use the settings url scheme instead as in `AppInfo.launch_default_for_uri ("settings://network", null);`